File: user_interface/joystick.py
import numpy as np
import math

class Joystick():
    
    MAX_TRIG_VAL = math.pow(2, 8)
    MAX_JOY_VAL = math.pow(2, 15)

    def __init__(self):
        self.BTN_EAST = 0 # X
        self.BTN_NORTH = 0 # N
        self.BTN_SOUTH = 0 # S
        self.BTN_WEST = 0 # W
        self.ABS_X = 0 
        self.ABS_Y = 0
        self.ABS_RY = 0
        self.ABS_RX = 0
        self.ABS_Z = 0 # LT
        self.ABS_RZ = 0 # RT

    def get_info(self, log=True):
        
        if log:
            print(self.BTN_EAST, self.BTN_NORTH, self.BTN_SOUTH, self.BTN_WEST, self.ABS_X, self.ABS_Y, self.ABS_RX, self.ABS_RY)

        return (self.BTN_EAST, self.BTN_NORTH, self.BTN_SOUTH, self.BTN_WEST, self.ABS_X, self.ABS_Y, self.ABS_RX, self.ABS_RY)
    def update(self, event):
        
        if event.code == 'SYN_REPORT':
            return 

        if event.code in ['ABS_X', 'ABS_Y', 'ABS_RX', 'ABS_RY']:
            setattr(self, event.code, round(event.state/Joystick().MAX_JOY_VAL, 3))
        elif event.code in ['ABS_Z', 'ABS_RZ']:
            setattr(self, event.code, round(event.state/Joystick().MAX_TRIG_VAL, 3))
        else:
            setattr(self, event.code, event.state)


        

File: master/Src/joystick.c
#include "joystick.h"
#include <math.h>

void Joystick_Init(void){
	my_joystick.BTN_EAST = 0; // X
	my_joystick.BTN_NORTH = 0; // N
  my_joystick.BTN_SOUTH = 0; // S
  my_joystick.BTN_WEST = 0; // W
  my_joystick.ABS_X=0.0;
  my_joystick.ABS_Y=0.0;
  my_joystick.ABS_RY=0.0;
  my_joystick.ABS_RX=0.0;
  my_joystick.ABS_Z=0.0; // LT
  my_joystick.ABS_RZ=0.0; // RT
}

void Joystick_SetValue(int id, float value){
	if(id==0)
		my_joystick.ABS_X = value;
	else if(id==1)
		my_joystick.ABS_Y = value;
	else if(id==2)
		my_joystick.ABS_RX = value;
	else if(id==3)
		my_joystick.ABS_RY = value;
	else if(id==4)
		my_joystick.ABS_Z = value;
	else if(id==5)
		my_joystick.ABS_RZ = value;
}